src/snowflake/connector/cursor.py

```diff
@@ -93,6 +94,19 @@
 ASYNC_RETRY_PATTERN = [1, 1, 2, 3, 4, 8, 10]
 INCIDENT_BLACKLIST = (KeyError, ValueError, TypeError)
 
+ResultMetadata = namedtuple(
+    "ResultMetadata",
+    [
+        "name",
+        "type_code",
+        "display_size",
+        "internal_size",
+        "precision",
+        "scale",
+        "is_nullable",
+    ],
+)
```

Comment:
You can actually add type hints to these by importing importing `NamedTuple` from `typing`.
For example see: https://docs.python.org/3/library/typing.html#typing.NamedTuple

Reply:
do you mean something like this?
```
ResultMetadata = NamedTuple(
    "ResultMetadata",
    [
        ("name", str),
        ("type_code", int),
        ("display_size", int),
        ("internal_size", int),
        ("precision", int),
        ("scale", int),
        ("is_nullable", bool),
    ],
)
```

Reply:
Good on you for using the backwards compatible implementation, but we don't actually support <3.6 so I went ahead and updated the definition to the non-backwards compatible one